Tribler/Test/Core/Modules/RestApi/test_downloads_endpoint.py

```diff
@@ -30,10 +30,105 @@ def verify_download(downloads):
             downloads_json = json.loads(downloads)
             self.assertEqual(len(downloads_json['downloads']), 2)
 
-        video_tdef, self.torrent_path = self.create_local_torrent(os.path.join(TESTS_DATA_DIR, 'video.avi'))
+        video_tdef, _ = self.create_local_torrent(os.path.join(TESTS_DATA_DIR, 'video.avi'))
         self.session.start_download_from_tdef(video_tdef, DownloadStartupConfig())
         self.session.start_download_from_uri("file:" + pathname2url(
             os.path.join(TESTS_DATA_DIR, "bak_single.torrent")))
 
         self.should_check_equality = False
         return self.do_request('downloads', expected_code=200).addCallback(verify_download)
+
+    @deferred(timeout=10)
+    def test_remove_download_no_remove_data_param(self):
+        """
+        Testing whether the API returns error 400 if the remove_data parameter is not passed
+        """
+        self.should_check_equality = False
+        return self.do_request('downloads/abcd/remove', expected_code=400, request_type='DELETE')
+
+    @deferred(timeout=10)
+    def test_remove_download_wrong_infohash(self):
+        """
+        Testing whether the API returns error 404 if a non-existent download is removed
+        """
+        self.should_check_equality = False
+        return self.do_request('downloads/abcd/remove', post_data={"remove_data": True},
+                               expected_code=404, request_type='DELETE')
+
+    @deferred(timeout=10)
+    def test_remove_download(self):
+        """
+        Testing whether the API returns 200 if a download is being removed
+        """
+        def verify_removed(_):
+            self.assertEqual(len(self.session.get_downloads()), 0)
+
+        video_tdef, _ = self.create_local_torrent(os.path.join(TESTS_DATA_DIR, 'video.avi'))
+        self.session.start_download_from_tdef(video_tdef, DownloadStartupConfig())
+        infohash = video_tdef.get_infohash().encode('hex')
+
+        return self.do_request('downloads/%s/remove' % infohash, post_data={"remove_data": True},
+                               expected_code=200, expected_json={"removed": True}, request_type='DELETE')\
+            .addCallback(verify_removed)
```

Comment:
To avoid the concatenation slash you could also do 
deferred = do_request...
return deferred.addCallback...

Just an idea.